### PR TITLE
Add auth token to StartServiceACK payload

### PR DESCRIPTION
**Pull request by jacobkeeler**
Partially implements #2215 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
TBD

### Summary
Added `authToken` field to StartServiceACK payload when connecting to cloud applications

##### Enhancements
* Adds `authToken` to StartServiceACK

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 1]
```diff
@@ -1909,15 +1920,18 @@ void PolicyHandler::OnSetCloudAppProperties(
...
   policy_manager_->InitCloudApp(policy_app_id);
 
+  bool auth_token_update = false;
   if (msg_params.keyExists(strings::enabled)) {
-    policy_manager_->SetCloudAppEnabled(policy_app_id,
-                                        msg_params[strings::enabled].asBool());
-
+    bool enabled = msg_params[strings::enabled].asBool();
+    policy_manager_->SetCloudAppEnabled(policy_app_id, enabled);
+    auth_token_update = enabled;
```
Comment (JackLivio) [inline src/components/application_manager/src/policies/policy_handler.cc]: Why should auth_token_update be set to enabled in this case?

File: src/components/protocol_handler/src/protocol_handler_impl.cc [hunk 2]
```diff
@@ -410,10 +411,22 @@ void ProtocolHandlerImpl::SendStartSessionAck(
...
                                                     kDisabledSecondary);
       }
+
+      std::string policy_app_id =
+          connection_handler_.GetCloudAppID(connection_id);
+      if (*min_version >= min_cloud_app_version && !policy_app_id.empty()) {
+        sync_primitives::AutoLock lock(auth_token_map_lock_);
+        auto it = auth_token_map_.find(policy_app_id);
+        if (it != auth_token_map_.end()) {
+          char auth_token[256];
```
Comment (JackLivio) [inline src/components/protocol_handler/src/protocol_handler_impl.cc]: authToken can be a lot longer that 256 chars.

```
      <param name="authToken" type="String" maxlength="65535" mandatory="false">
            <description>Used to authenticate websocket connection on app activation</description>
        </param>
```

File: src/components/application_manager/src/policies/policy_handler.cc (hunk 1 above)
Comment (jacobkeeler) [reply src/components/application_manager/src/policies/policy_handler.cc]: Updates are sent for all enabled apps when the policy table is loaded or updated. This logic is to cover the case where an app is enabled after the policy table is initially loaded, meaning that the listener does not already have the auth token for this app.

File: src/components/protocol_handler/src/protocol_handler_impl.cc (hunk 2 above)
Comment (jacobkeeler) [reply src/components/protocol_handler/src/protocol_handler_impl.cc]: Good catch

Comment (jacobkeeler) [reply src/components/protocol_handler/src/protocol_handler_impl.cc]: Resolved in latest comment

**Pull request merged**